File: Assignment28_4.java
import java.util.Scanner;

////////////////////////////////////////////////////////////////////////////
//    Function Name :       Pattern
//    Description :         Display reverse row-number pattern
//    Input :               Integer
//    Output :              Prints the pattern
//    Author :              Raj Samir Jadhav
//    Date :                24/11/2025
////////////////////////////////////////////////////////////////////////////

class Pattern
{
    public void Display(int iRow, int iCol)      // Method to print pattern
    {
        for(int i = iRow; i >= 1; i--)           // Start from iRow down to 1
        {
            for(int j = 1; j <= iCol; j++)       // Loop for columns
            {
                System.out.print(i + "\t");      // Print row number
            }
            System.out.println();
        }
    }
}

class Assignment28_4
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0, iValue2 = 0;

        System.out.println("Enter number of rows : ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter number of columns : ");
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Display(iValue1, iValue2);          // Method call
    }
}

File: Assignment28_4.c
#include <stdio.h>

///////////////////////////////////////////////////////////////////////////
//    Function Name :       Pattern
//    Description :         Display reverse row-number pattern
//    Input :               Integer
//    Output :              Prints the pattern
//    Author :              Raj Samir Jadhav
//    Date :                24/11/2025
////////////////////////////////////////////////////////////////////////////

void Pattern(int iRow, int iCol)        // Function to display the patter
{
    int i = 0, j = 0;

    for(i = iRow; i >= 1; i--)          // Start from last row → iRow down to 1
    {
        for(j = 1; j <= iCol; j++)      // Loop for columns
        {
            printf("%d\t", i);          // Print same row number in each column
        }
        printf("\n");                   // New line after each row
    }
}

int main()
{
    int iValue1 = 0, iValue2 = 0;

    printf("Enter number of rows and columns : ");
    scanf("%d %d", &iValue1, &iValue2);

    Pattern(iValue1, iValue2);          // Function call

    return 0;
}
